presto-tests/src/test/java/com/facebook/presto/tests/TestSqlFunctions.java

```diff
@@ -48,4 +48,15 @@ public void testCreateFunctionInvalidSemantics()
                 "CREATE FUNCTION testing.default.tan (x double) RETURNS double COMMENT 'tangent trigonometric function' RETURN sum(x)",
                 ".*CREATE FUNCTION body cannot contain aggregations, window functions or grouping operations:.*");
     }
+
+    @Test
+    public void testDropFunctionInvalidFunctionName()
+    {
+        assertQueryFails(
+                "DROP FUNCTION IF EXISTS testing.tan",
+                ".*Invalid function name: testing\\.tan, require exactly 3 parts");
```

Comment:
Maybe change the error message to be something more instructive like "function name should be in the form of catalog.schema.function_name"? I'm not sure users would understand what does "require exactly 3 parts" mean.